scripts/ci/pre_commit/pre_commit_sync_dag_init_decorator.py

```diff
@@ -42,8 +42,10 @@ def _find_dag_deco(mod: ast.Module) -> ast.FunctionDef:
     return next(n for n in ast.iter_child_nodes(mod) if isinstance(n, ast.FunctionDef) and n.name == "dag")
 
 
+import ast
+
 # The new unparse() output is much more readable; fallback to dump() otherwise.
-if sys.version_info >= (3, 8):
+if sys.version_info >= (3, 8) and hasattr(ast, 'unparse'):
```

Comment:
I thnk the check is just wrong. According to https://docs.python.org/3.9/library/ast.html#ast.unparse it should be >= (3,9)